pkg/controller/operators/catalog/operator.go

```diff
@@ -1064,23 +1064,41 @@ func (o *Operator) ResolvePlan(plan *v1alpha1.InstallPlan) error {
 	return nil
 }
 
-// Ensure all existing versions are present in new CRD
+func getCRDVersionsMap(crd *v1beta1ext.CustomResourceDefinition) map[string]struct{} {
+	versionsMap := map[string]struct{}{}
+
+	for _, version := range crd.Spec.Versions {
+		versionsMap[version.Name] = struct{}{}
+	}
+	if crd.Spec.Version != "" {
+		versionsMap[crd.Spec.Version] = struct{}{}
+	}
+
+	return versionsMap
+}
+
+// Ensure all existing served versions are present in new CRD
 func ensureCRDVersions(oldCRD *v1beta1ext.CustomResourceDefinition, newCRD *v1beta1ext.CustomResourceDefinition) error {
+	newCRDVersions := getCRDVersionsMap(newCRD)
+
 	for _, oldVersion := range oldCRD.Spec.Versions {
-		var versionPresent bool
-		for _, newVersion := range newCRD.Spec.Versions {
-			if oldVersion.Name == newVersion.Name {
-				versionPresent = true
+		if oldVersion.Served {
+			_, ok := newCRDVersions[oldVersion.Name]
+			if !ok {
+				return fmt.Errorf("New CRD (%s) must contain existing served versions (%s)", oldCRD.Name, oldVersion.Name)
 			}
 		}
-		if !versionPresent {
-			return fmt.Errorf("not allowing CRD (%v) update with unincluded version %v", newCRD.GetName(), oldVersion)
+	}
+	if oldCRD.Spec.Version != "" {
```

Comment:
Shouldn't the served boolean be checked here too?

Reply:
The `version` field is just a string. There is no served field.